### ISSUE TITLE
Add link to owncloud web version in app

### ISSUE THREAD

**Issue by subiol** [nextcloud/android#356]
It would be convinient to have a link in the app that launched a web browser with the user automatically logged in in the ownclud web page.

**Comment by AndyScherzinger** [comment]
Hi @subiol can you go a little more into details on this request?

* do you simply want to open the top folder in the web browser or
* open the folder you have currently browsed to in the Android client
* what would be the use case for launching the browser: If you need to open the browser because you are missing a function offered by the web front-end but not the Android client, then we should discuss integrating it :)

**Comment by tobiasKaminsky** [comment]
Also one question from me: do you have a suggestion where to place this? In the overflow menu called "open in browser" maybe?

ps: probably you mean nextcloud instead of owncloud ;-)

**Comment by subiol** [comment]
@Andy The idea of accessing the web page is to access other functions/apps like bookmarks f.e. or any other plugin. I have no complaints about the file syncing functionalities of the app.

@Tobias Just under settings over participate? But I'm not a gui guy so there might be a better option.

> probably you mean nextcloud instead of owncloud ;-)

Did I wrote owncloud? I guess old ways die hard.

**Comment by tobiasKaminsky** [comment]
We can enhance this in the way that we detect the additional plugins, like bookmarks and then place them in the drawer and open them directly with a web view.

**Comment by subiol** [comment]
Sounds great

**Comment by mario** [comment]
No, we can't @tobiasKaminsky. That would require us to log-in instead of the user, and we will of course NOT do that.

**Comment by feutl** [comment]
I have given my family a Nextcloud and looking at the parents working with it I figured out they are overwhelmed and unable to understand why they need a nextcloud app and a browser and remeber URLs or find bookmarks.

It would be therefor great to have the app, login in there and than the nextcloud app lets you open the browser directly from within the app or even better, opens a browser WITHIN the app (like feed readers are doing).

My idea would have been to have the same "apps" being shown in the nextcloud app. In my case would be contacts, calendar, mail and files. The all work great already with the mobile view on a mobile device but it is just not seemles for people to get there. A click on an icon an the web version opens, would be great.

This would allow to have nextcloud as THE SINGLE app to get access to everything. Files natively supported, everything else not yet "natively" implemented in the browser (within or external app).

**Comment by rullzer** [comment]
Currently this is not possible. The apptoken the android client has can't be used to login to the web.

**Comment by tobiasKaminsky** [comment]
@rullzer are there any plans to change/enhance this?
Opening apps, like mail, dashboard, … in an internal webview with logged in user would be nice.
Using the/one apptoken would be security wise fine, in my opinion.

**Comment by mario** [comment]
Personally, I would not encourage this.

**Comment by feutl** [comment]
@rullzer @tobiasKaminsky 
It would be fine, for a first stage deployment to provide the credentials manually and "keep the user logged" in as a webbrowser would do

@mario 
why? I mean keeping thing in "one" app makes it easier for everyone to adopt and use this one app. I have 3 persons already in my family who would find that much easier than starting a browser or configure another application, and all 3 of them are far away from techies.

**Comment by tobiasKaminsky** [comment]
@feutl for this, you then can simply use external sites https://apps.nextcloud.com/apps/external.
This one allows you to create a link in the drawer menu and open it in a webview.

**Comment by feutl** [comment]
this has nothing to do with the webview, that one is ok.
but if someone uses a tablet/phone opens the nextcloud app, there is no seamless way to get from the nextcloud app (where only files are handled) to the nextcloud mail app, or news app or contacts app. you need to open a browser and open the URL to the website and click on the icon in the menu bar.

this could be easier for someone who is not a techie, like my mum for example ;)

**Comment by feutl** [comment]
@tobiasKaminsky 
I am sorry, I take everything back, I haven't seen the option, was blind, or just not focuses enough.
This is perfect for what I need at the moment.

Still seamless login integration, and automatically offering the apps already activated but not the Android/iOS App would be great too.

Thanks and Sorry

**Comment by mr-bolle** [comment]
@mario I also find this useful, there are many apps for Nextcloud available. These will certainly not be so easily integrated into the app (collabora office, deck, music player ...)
Not every user wants to enter / save the credentials in the browser.

Is there a possibility to pass a unique link to the browser (as currently solved in the dev version 'stream with')?

**Comment by R7JANV1** [comment]
Yes it would save a lot of support effort to have a menu in the app that opens an in app browser or something similar.
